packages/nimble-components/src/table-column/base/types.ts

```diff
@@ -32,6 +32,16 @@ export const TableColumnSortOperation = {
 export type TableColumnSortOperation =
     (typeof TableColumnSortOperation)[keyof typeof TableColumnSortOperation];
 
+/**
+ * The possible selection states that the table or a table row can be in.
+ */
+export const TableRowSelectionState = {
```

Comment:
Two notes about this enum:
1. It is internal, so I put it here with the other internal types. It isn't used by columns, so this might not be the right home for it.
2. I made this an enum rather than a boolean because it will be extended to include a partial/indeterminate state when multi-select support is added.

Reply:
Making it `@internal` within `table/types.ts` seems reasonable to me. I'll move it.